hack/tools/e2e-log-sync/main.go

```diff
@@ -0,0 +1,268 @@
+//go:build tools
+// +build tools
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/pkg/errors"
+	"github.com/valyala/fastjson"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	"k8s.io/klog/v2"
+)
+
+var (
+	bucket           = flag.String("log-bucket", "kubernetes-jenkins", "Bucket to download the logs from")
+	controllerFolder = flag.String("log-controller-folder", "", "Folder to get the controller-logs from")
```

Comment:
Those two parameters have to be taken from the ProwJob UI (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/6189/pull-cluster-api-e2e-main/1496233284105932800)
(although the bucket should be always `kubernetes-jenkins`)

We could abstract away the GCS details by providing flags for:
* org/repo
* PR number (or periodic)
* job 
* build id (if not set => latest)

But maybe the current variant is more user-friendly

Reply:
One flag for convenience could be to make it possible to just hand over the link to the ProwJob: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/6189/pull-cluster-api-e2e-main/1496233284105932800 
We can infer everything from there

Reply:
Done